packages/lexical-code/src/__tests__/unit/LexicalCodeNodeTabs.test.ts

```diff
@@ -0,0 +1,222 @@
+          await editor.update(() => {
+            const root = $getRoot();
+            codeNode = $createCodeNode();
+            root.append(codeNode);
+            codeNode.selectStart();
+            $getSelection()!.insertRawText(getRawText(scenario[0]));
+          });
+
+          // Create a RangeSelection that matches the selection
+          // described in the scenario input via the | convention
+          await editor.update(() => {
+            const selection = $createRangeSelection();
+            if ($isCodeNode(codeNode) && codeNode.isEmpty()) {
+              selection.anchor.set(codeNode.getKey(), 0, 'element');
+              selection.focus.set(codeNode.getKey(), 0, 'element');
+            } else {
+              const rawText = getRawTextWithSelection(scenario[0]);
+              const selFirst = rawText.indexOf('|');
+              let selLast = rawText.lastIndexOf('|');
+
+              // if selection is not collapsed, we need to adjust
+              // the last index to take account of the first |
+              if (selLast !== selFirst) {
+                selLast -= 1;
+              }
+
+              let matching: null | LineBreakNode | TabNode | CodeHighlightNode =
+                codeNode.getFirstChild();
+              let parentIndex = 0;
+              let offset = 0;
+              while (
+                matching !== null &&
+                offset + matching.getTextContentSize() < selFirst
+              ) {
+                offset += matching.getTextContentSize();
+                matching = matching.getNextSibling();
+                parentIndex++;
+              }
+              if (matching === null) {
+                selection.anchor.set(codeNode.getKey(), 0, 'element');
+              } else if (!$isLineBreakNode(matching)) {
+                selection.anchor.set(
+                  matching.getKey(),
+                  selFirst - offset,
+                  'text',
+                );
+              } else {
+                selection.anchor.set(codeNode.getKey(), parentIndex, 'element');
+              }
+              while (
+                matching !== null &&
+                offset + matching.getTextContentSize() < selLast
+              ) {
+                offset += matching.getTextContentSize();
+                matching = matching.getNextSibling();
+                parentIndex++;
+              }
+              if (matching === null) {
+                selection.focus.set(codeNode.getKey(), 0, 'element');
+              } else if (!$isLineBreakNode(matching)) {
+                selection.focus.set(
+                  matching.getKey(),
+                  selLast - offset,
+                  'text',
+                );
+              } else {
+                selection.focus.set(
+                  codeNode.getKey(),
+                  parentIndex + 1,
+                  'element',
+                );
+              }
+            }
+            $setSelection(selection);
```

Comment:
```suggestion
            $setSelectionFromCaretRange(
              $getCaretRangeInDirection(
                $caretRangeFromSelection(selection),
                direction === 'forwards' ? 'next' : 'previous',
              ),
            );
```

This could also be simplified if PointCaret was used for anchor/focus instead of manipulating a RangeSelection.